torchtitan/config_manager.py

```diff
@@ -502,6 +502,13 @@ class FaultTolerance:
     min_replica_size: int = 1
     """The minimum number of FT replica for each step."""
 
+    training_method: str | None = "diloco"
```

Comment:
`semi_sync_method` or `synchronize_method`  are less confusing. `trianing_method` is pretty general. And can we specify that if the value is not set, we will use synchronized training?

Reply:
`semi_sync_method` sounds good. I will also update that comment